shell/platform/android/test/io/flutter/embedding/engine/systemchannels/PlatformChannelTest.java

```diff
@@ -113,6 +115,61 @@ public void setSystemExclusionRectsSendsJSONExceptionOnIncorrectDataShape() thro
         );
     }
 
+    @Test
+    public void getSystemExclusionRectsSendsExclusionRectsToFrameworkOnSuccess() throws JSONException {
+        DartExecutor dartExecutor = mock(DartExecutor.class);
+        PlatformChannel platformChannel = new PlatformChannel(dartExecutor);
+        PlatformMessageHandler platformMessageHandler = mock(PlatformMessageHandler.class);
+        platformChannel.setPlatformMessageHandler(platformMessageHandler);
+
+        int top = 0;
+        int right = 500;
+        int bottom = 250;
+        int left = 0;
+
+        ArrayList<Rect> expectedExclusionRects = new ArrayList<Rect>();
+        Rect gestureRect = new Rect(left, top, right, bottom);
+        expectedExclusionRects.add(gestureRect);
+        when(platformMessageHandler.getSystemGestureExclusionRects()).thenReturn(expectedExclusionRects);
+
+        MethodCall callGetSystemGestureExclusionRects = new MethodCall(
+            "SystemGestures.getSystemGestureExclusionRects",
+            null
+        );
+        ResultsMock resultsMock = mock(ResultsMock.class);
+        platformChannel.parsingMethodCallHandler.onMethodCall(callGetSystemGestureExclusionRects, resultsMock);
+
+        ArrayList<HashMap<String, Integer>> expectedEncodedOutputRects = new ArrayList<HashMap<String, Integer>>();
+        HashMap<String, Integer> rectMap = new HashMap<String, Integer>();
+        rectMap.put("top", top);
+        rectMap.put("right", right);
+        rectMap.put("bottom", bottom);
+        rectMap.put("left", left);
+        expectedEncodedOutputRects.add(rectMap);
+        verify(resultsMock, times(1)).success(expectedEncodedOutputRects);
+    }
+
+    @Test
+    public void getSystemExclusionRectsSendsIncorrectAPILevelException() {
+        DartExecutor dartExecutor = mock(DartExecutor.class);
+        PlatformChannel platformChannel = new PlatformChannel(dartExecutor);
+        PlatformMessageHandler platformMessageHandler = mock(PlatformMessageHandler.class);
+        platformChannel.setPlatformMessageHandler(platformMessageHandler);
+        when(platformMessageHandler.getSystemGestureExclusionRects()).thenReturn(null);
+
+        MethodCall callGetSystemGestureExclusionRects = new MethodCall(
+            "SystemGestures.getSystemGestureExclusionRects",
+            null
+        );
+        ResultsMock resultsMock = mock(ResultsMock.class);
+        platformChannel.parsingMethodCallHandler.onMethodCall(callGetSystemGestureExclusionRects, resultsMock);
```

Comment:
Please clearly separate the test setup from the behavior under test from the verification.